src/main/java/sirius/pasta/noodle/sandbox/Sandbox.java

```diff
@@ -96,21 +98,64 @@ public static boolean isAccessGranted(NoodleSandbox annotation) {
 
     /**
      * Determines if the given method can be invoked by user code.
+     * <p>
+     * We search up the inheritance tree for the first method that is either allowed or rejected via
+     * {@link NoodleSandbox annotation} or via config. The approach however is simplified and doesn't support
+     * interfaces.
      *
      * @param method the method to check
      * @return <tt>true</tt> if it can be invoked, <tt>false</tt> otherwise
      */
     public boolean canInvoke(Executable method) {
-        return isControlledViaAnnotation(method).orElseGet(() -> isAllowed(method.getDeclaringClass().getName(),
-                                                                           method.getName()));
+        Executable currentMethod = method;
+
+        while (currentMethod != null) {
+            Optional<Boolean> controlled = isControlled(currentMethod);
+
+            if (controlled.isPresent()) {
+                return controlled.get();
+            }
+
+            currentMethod = getSuperclassMethod(currentMethod);
+        }
+
+        return false;
+    }
+
+    private Method getSuperclassMethod(Executable method) {
+        Class<?> superclass = method.getDeclaringClass().getSuperclass();
+
+        if (superclass == null) {
+            return null;
+        }
+
+        try {
+            return superclass.getMethod(method.getName(), method.getParameterTypes());
+        } catch (NoSuchMethodException e) {
+            Exceptions.ignore(e);
+            return null;
+        }
```

Comment:
what if an intermediate class doesn't provide the method? will this still work or does getMethod only provide the methods defined of the class itself

Reply:
It works as intended, getMethod finds the first match in the inheritance tree